### ISSUE TITLE
Add Object.merge

### ISSUE THREAD

**Issue by zackify** [Legitcode/rubyfill#5]
If Object.assign is available use this, otherwise make a polyfill and use that. @rampage1213

**Comment by dphaener** [comment]
Danger. Using other polyfills in a polyfill is a bad idea. Especially since you have the option to include only one polyfill.

**Comment by zackify** [comment]
no im saying create a polyfill if object.assign can't be used. not to include another person's

**Comment by dphaener** [comment]
Yes I know. But what if someone tries to use this polyfill on its own and it requires another one of our polyfills? Require it? Maybe. Don't know if I like that.

**Comment by zackify** [comment]
I was thinking of something like: if object.assign return object.assign({},
this, newObject) else loop through and merge the old way. Not actually
including something else

**Comment by dphaener** [comment]
Gotcha.
